### PR TITLE
Updates and corrections to the ARIA radio doc

### PR DESCRIPTION
**Pull request by scottaohara**
Upon reviewing the merged #17924, I noticed some additional typos in some of the markup examples. Those have been fixed in this PR.

Additionally, there were some instances of inaccurate information which have now also been corrected.
For instance: 
- removes the mention of using `aria-checked=mixed`. This is not valid for radio buttons.
- revises the section on how to indicate a required radio group. ARIA does not presently allow for the use of `aria-required` on individual `role=radio` elements. 
- changed the mention of `tabindex` in the Associated WAI-ARIA Roles, States, and Properties section to be a note - because it's not an ARIA attribute, so it didn't make sense to include it in the listing of ARIA attributes.
- Changed the examples to use `<div>` elements to group the radio/labels, rather than list items.  The use of list items would add extra enumeration to the radio buttons.  Being associated with each other, the radio buttons already announce an x of y... so the list items just add noise.
- Removed the mention of `onFocus` and `onBlur` from the required JavaScript section.  They had nothing written about them, so it didn't make sense for them to be there.

### CODE REVIEW

File: files/en-us/web/accessibility/aria/roles/radio_role/index.md [hunk 1]
```diff
@@ -8,67 +8,63 @@ tags:
   - ARIA widget
   - Reference
 ---
-The `radio` role is one of a group of checkable buttons, in a `radiogroup`, where no more than one of the buttons can be checked at a time.
+The `radio` role is one of a group of checkable radio buttons, in a `radiogroup`, where no more than a single radio button can be checked at a time.
```
Comment (hidde) [inline files/en-us/web/accessibility/aria/roles/radio_role/index.md:11]: Much clearer!

File: files/en-us/web/accessibility/aria/roles/radio_role/index.md [hunk 2]
```diff
@@ -8,67 +8,63 @@ tags:
...
+    <input type="radio" value="huh?" id="q25_radio3"  name="q25" checked>
+    <label for="q25_radio3">What is the question?</label>
+  </div>
 </fieldset>
 ```
 
-The native HTML radio ([`<input type="radio">`](/en-US/docs/Web/HTML/Element/input/radio)) form control had two states ("checked" or "not checked"). Similarly, an element with `role="radio"` can expose two states through the [`aria-checked`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-checked) attribute: `true` and `false`. The value of `mixed` correlates to the  `indeterminate` value of HTML `checked` attribute, meaning neither checked nor unchecked. While `mixed` is valid on both radios and checkboxes, it is rarely useful for a radio button.
+The native HTML radio ([`<input type="radio">`](/en-US/docs/Web/HTML/Element/input/radio)) form control has two states ("checked" or "not checked"). Similarly, an element with `role="radio"` can expose two states through the [`aria-checked`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-checked) attribute: `true` representing the checked state, and `false` representing the unchecked state. The `aria-checked` value of `mixed` is not valid to use for a radio button.
```
Comment (hidde) [inline files/en-us/web/accessibility/aria/roles/radio_role/index.md:57]: Tiny nit: maybe here we want 'form control' before the brackets / link to input type radio?

```suggestion
The native HTML radio form control ([`<input type="radio">`](/en-US/docs/Web/HTML/Element/input/radio)) has two states ("checked" or "not checked"). Similarly, an element with `role="radio"` can expose two states through the [`aria-checked`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-checked) attribute: `true` representing the checked state, and `false` representing the unchecked state. The `aria-checked` value of `mixed` is not valid to use for a radio button.
```

File: files/en-us/web/accessibility/aria/roles/radio_role/index.md [hunk 3]
```diff
@@ -8,67 +8,63 @@ tags:
...
 
-If any of the radio roles in a group has `aria-required="true"` set, it is as if all of the radios in the group had the attribute making the selection of one of the radios in the radiogroup being required to be valid;  though not necessarily the one that has the [`aria-required`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-required) attribute set.
+To programmatically indicate that a radio button must be chosen from a radio group, then the [`aria-required`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-required) attribute, with a value of `true`, must be specified on the `radiogroup` element. It is not expected to use the `aria-required` attribute on individual ARIA radio buttons.
```
Comment (hidde) [inline files/en-us/web/accessibility/aria/roles/radio_role/index.md:67]: Nit: it seems ‘then’ doesn't work with ‘to’; maybe remove 'then', or say something like ‘If we want to programmatically indicate’?

**Review by hidde — APPROVED** [review]
Thanks for all the added clarity. Added a few nitpicky comments, let me know what you think and I'll be happy to merge either way!

File: files/en-us/web/accessibility/aria/roles/radio_role/index.md [hunk 4]
```diff
@@ -8,67 +8,63 @@ tags:
...
```
Comment (scottaohara) [inline files/en-us/web/accessibility/aria/roles/radio_role/index.md:67]:
```suggestion
To programmatically indicate that a radio button must be chosen from a radio group the [`aria-required`](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-required) attribute, with a value of `true`, must be specified on the `radiogroup` element. It is not expected to use the `aria-required` attribute on individual ARIA radio buttons.  
```

**Comment by hidde** [comment]
Thanks for these changes!

**Pull request merged**